hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/keyvalue/TestKeyValueContainerMetadataInspector.java

```diff
@@ -290,11 +402,53 @@ private void checkJsonErrorsReport(JsonObject jsonReport,
 
   public void setDBBlockAndByteCounts(KeyValueContainerData containerData,
       long blockCount, long byteCount) throws Exception {
+    setDB(containerData, blockCount, byteCount,
+        0, Collections.emptyList());
+  }
+
+  public void setDB(KeyValueContainerData containerData,
+      long blockCount, long byteCount,
+      long dbDeleteCount, List<DeletedBlocksTransaction> deleteTransactions)
+      throws Exception {
     try (DBHandle db = BlockUtils.getDB(containerData, getConf())) {
       Table<String, Long> metadataTable = db.getStore().getMetadataTable();
       // Don't care about in memory state. Just change the DB values.
       metadataTable.put(containerData.getBlockCountKey(), blockCount);
       metadataTable.put(containerData.getBytesUsedKey(), byteCount);
+      metadataTable.put(containerData.getPendingDeleteBlockCountKey(),
+          dbDeleteCount);
+
+      final DatanodeStore store = db.getStore();
+      LOG.info("store {}", store.getClass().getSimpleName());
+      if (store instanceof DatanodeStoreSchemaTwoImpl) {
```

Comment:
Why do we need a split between schema v2 and v3 here? Looks like they do the same thing. `containerData.getDeleteTxnKey(t.getTxID())` should work for both schemas.

Reply:
There are `DatanodeStoreSchemaTwoImpl` and `DatanodeStoreSchemaThreeImpl`.  The key types for `delTxTable` are different  in v2 and v3.